x-pack/test/detection_engine_api_integration/security_and_spaces/tests/get_rule_execution_events.ts

```diff
@@ -0,0 +1,227 @@
+      expect(response.body.events[0].search_duration_ms).to.eql(0);
+      expect(response.body.events[0].schedule_delay_ms).to.greaterThan(0);
+      expect(response.body.events[0].indexing_duration_ms).to.eql(0);
+      expect(response.body.events[0].gap_duration_ms).to.eql(0);
+      expect(response.body.events[0].security_status).to.eql('partial failure');
+      expect(
+        response.body.events[0].security_message.startsWith(
+          'Check privileges failed to execute ResponseError: index_not_found_exception: [index_not_found_exception] Reason: no such index [no-name-index]'
+        )
+      ).to.eql(true);
+    });
+
+    // TODO: Debug indexing
+    it.skip('should return execution events for a rule that has executed in a failure state with a gap', async () => {
+      const rule = getRuleForSignalTesting(['auditbeat-*'], uuid.v4(), false);
+      const { id } = await createRule(supertest, log, rule);
+
+      const start = dateMath.parse('now')?.utc().toISOString();
+      const end = dateMath.parse('now+24h', { roundUp: true })?.utc().toISOString();
+
+      // Create 5 timestamps a minute apart to use in the templated data
+      const dateTimes = [...Array(5).keys()].map((i) =>
+        moment(start)
+          .add(i + 1, 'm')
+          .toDate()
+          .toISOString()
+      );
+
+      const events = failedGapExecution.map((e, i) => {
+        set(e, '@timestamp', dateTimes[i]);
+        set(e, 'event.start', dateTimes[i]);
+        set(e, 'event.end', dateTimes[i]);
+        set(e, 'rule.id', id);
+        return e;
+      });
+
+      await indexEventLogExecutionEvents(es, log, events);
+      await waitForEventLogExecuteComplete(es, log, id);
+
+      const response = await supertest
+        .get(detectionEngineRuleExecutionEventsUrl(id))
+        .set('kbn-xsrf', 'true')
+        .query({ start, end });
+
+      // console.log(JSON.stringify(response));
+
+      expect(response.status).to.eql(200);
+      expect(response.body.total).to.eql(1);
+      expect(response.body.events[0].duration_ms).to.eql(4236);
+      expect(response.body.events[0].search_duration_ms).to.eql(0);
+      expect(response.body.events[0].schedule_delay_ms).to.greaterThan(0);
+      expect(response.body.events[0].indexing_duration_ms).to.eql(0);
+      expect(response.body.events[0].gap_duration_ms).to.greaterThan(0);
+      expect(response.body.events[0].security_status).to.eql('failed');
+      expect(
+        response.body.events[0].security_message.startsWith(
+          'Check privileges failed to execute ResponseError: index_not_found_exception: [index_not_found_exception] Reason: no such index [no-name-index]'
+        )
+      ).to.eql(true);
+    });
+
+    // it('should return execution events when providing a status filter', async () => {
+    //   const rule = getRuleForSignalTesting(['auditbeat-*', 'no-name-index']);
+    //   const { id } = await createRule(supertest, log, rule);
```

Comment:
Working through these remaining functional tests now -- adding templated event-log events for exercising gaps (to deal with timing/RBAC) and additional tooling around checking event-log state. We don't have any existing API integration helpers with event-log we can use, no? Are there specific alerting helpers we could leverage here as well?